perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTest.java

```diff
@@ -9839,6 +9839,55 @@ public void setAttributeRights() throws Exception {
 		}
 	}
 
+	@Test
+	public void getAttributePolicyCollections() throws Exception {
+		System.out.println(CLASS_NAME + "getAttributePolicyCollections");
+
+		Attribute groupAttribute1 = setUpAttribute(String.class.getName(), "testGroupAttribute1", AttributesManager.NS_GROUP_ATTR_DEF, "TEST VALUE");
+
+		List<AttributePolicyCollection> policyCollections = new ArrayList<>();
+		List<AttributePolicy> policies = new ArrayList<>();
+
+		policies.add(new AttributePolicy(123, Role.GROUPADMIN, RoleObject.Group, 1));
+		policies.add(new AttributePolicy(789, Role.PERUNOBSERVER, RoleObject.None, 1));
+		policyCollections.add(new AttributePolicyCollection(42, groupAttribute1.getId(), AttributeAction.READ, new ArrayList<>(policies)));
+
+		policies.clear();
+		policies.add(new AttributePolicy(123, Role.GROUPADMIN, RoleObject.Group, 1));
+		policyCollections.add(new AttributePolicyCollection(43, groupAttribute1.getId(), AttributeAction.WRITE, new ArrayList<>(policies)));
+
+		perun.getAttributesManager().setAttributePolicyCollections(sess, policyCollections);
+
+		List<AttributePolicyCollection> insertedGroupCollections = perun.getAttributesManager().getAttributePolicyCollections(sess, groupAttribute1.getId());
+		List<AttributePolicy> insertedPolicies;
+
+		assertEquals(2, insertedGroupCollections.size());
+		assertEquals(groupAttribute1.getId(), insertedGroupCollections.get(0).getAttributeId());
```

Comment:
You can move this check into the forcycle and check apc.getId() for both collections there.